### PR TITLE
Add remove opportunity broadcast for expired opportunities

### PR DESCRIPTION
**Pull request by danimhr**
This PR aims to broadcast remove opportunity message for those opportunities that are expired.

### CODE REVIEW

File: auction-server/src/opportunity/repository/remove_opportunity.rs [hunk 1]
```diff
@@ -17,8 +20,9 @@ impl<T: InMemoryStore> Repository<T> {
         &self,
         db: &sqlx::Pool<Postgres>,
         opportunity: &T::Opportunity,
-        reason: OpportunityRemovalReason,
+        reason: entities::OpportunityRemovalReason,
```
Comment (m30m) [inline auction-server/src/opportunity/repository/remove_opportunity.rs:23]: maybe `reason: impl Into<OpportunityRemovalReason>,` ? if you just want it to be convertable

File: auction-server/src/opportunity/service/remove_invalid_or_expired_opportunities.rs [hunk 2]
```diff
@@ -59,16 +66,37 @@ where
...
+                        .remove_opportunity(&self.db, opportunity, reason)
                         .await
                     {
-                        tracing::error!("Failed to remove opportunity: {}", e);
+                        Ok(()) => {
+                            if self
+                                .repo
+                                .get_in_memory_opportunities_by_key(&opportunity.get_key())
+                                .await
+                                .is_empty()
```
Comment (m30m) [inline auction-server/src/opportunity/service/remove_invalid_or_expired_opportunities.rs:83]: I think this logic deserves a few lines of comment

**Review by m30m — COMMENTED** [review]
I don't think we need the evm variations for now. Let's discuss this today

File: auction-server/src/opportunity/repository/remove_opportunity.rs (hunk 1 above)
Comment (danimhr) [reply auction-server/src/opportunity/repository/remove_opportunity.rs:23]: I want to get the removal reason type of the entities :-?
We should use entities or primary types in repo functions

**Pull request merged**